src/components/Dialogs/FittingDialog/FittingDialogComponent.tsx

```diff
@@ -112,67 +149,110 @@ export class FittingDialogComponent extends React.Component {
-                    </FormGroup>
-                    <FormGroup label="Components" inline={true}>
-                        <SafeNumericInput className="components-input" value={fittingStore.components.length} min={1} max={20} stepSize={1} onValueChange={val => fittingStore.setComponents(Math.round(val))} />
-                        {fittingStore.components.length > 1 && (
-                            <>
-                                <Slider
-                                    value={fittingStore.selectedComponentIndex + 1}
+                </div>
+                <SplitPane split="horizontal" defaultSize="50%">
+                    <div className="upper-pane">
+                        <div className={classNames(Classes.DIALOG_BODY, "unpinned-input-panel")}>
+                            <FormGroup label="Region" inline={true}>
+                                <HTMLSelect value={fittingStore.selectedRegionId} options={fittingStore.regionOptions} onChange={ev => fittingStore.setSelectedRegionId(parseInt(ev.target.value))} />
+                            </FormGroup>
+                            <FormGroup label="Components" inline={true}>
+                                <SafeNumericInput
+                                    className="components-input"
+                                    selectAllOnFocus={true}
+                                    value={fittingStore.components.length}
                                     min={1}
+                                    max={20}
                                     stepSize={1}
-                                    max={fittingStore.components.length}
-                                    showTrackFill={false}
-                                    onChange={val => fittingStore.setSelectedComponentIndex(val - 1)}
-                                    disabled={fittingStore.components.length <= 1}
+                                    onValueChange={val => setTimeout(() => fittingStore.setComponents(Math.round(val)), 0)} // wait for onBlur events of the inputs
                                 />
-                                <Tooltip2 content="Delete current component">
-                                    <AnchorButton icon={"trash"} onClick={fittingStore.deleteSelectedComponent} />
+                                {fittingStore.components.length > 1 && (
+                                    <>
+                                        <Slider
+                                            value={fittingStore.selectedComponentIndex + 1}
+                                            min={1}
+                                            stepSize={1}
+                                            max={fittingStore.components.length}
+                                            showTrackFill={false}
+                                            onChange={val => setTimeout(() => fittingStore.setSelectedComponentIndex(val - 1), 0)} // wait for onBlur events of the inputs
```

Comment:
is there no better way to handle this? :thinking:

Reply:
@veggiesaurus Another way is to update the inputs with all keydown events instead of onBlur and enter key events. Currently I am reusing `CoordNumericInput`, so it's not easy. I will add a todo here.